src/connection.rs

```diff
@@ -49,9 +51,12 @@ pub struct HttpConnection<T> {
     /// A buffer containing the bytes of a response that is currently
     /// being sent.
     response_buffer: Option<Vec<u8>>,
+    /// The latest file that has been received and which must be associated
+    /// with the pending request.
+    file: Option<File>,
```

Comment:
nit: rename for clarity

```suggestion
    /// The latest file that has been received and which must be associated
    /// with the pending request.
    received_file: Option<File>,
```
or smth similar to suggest this is a resource coming in through the `HttpConnection` and not a property of it.